### DISCUSSION TITLE
How can I limit scan area?

### DISCUSSION THREAD

**Discussion by ssynowiec** [expo/expo#26907] [Archived: Questions]
Hey 👋

I'm using `CameraView` from the `expo-camera/next` and I implemented barcodes scanner, but i want to limit scanning area to centered square with border, how can i do?

Expo version: 50.0.4

**Comment by DudeFactory** [comment]
Limit barcode section is good, but maybe at least we can make camera view 1:1 (square)?

**Comment by shel404** [comment]
```JS
  const isWithinScannableArea = (
    cornerPoints: Point[],
    scannableArea: { x: number; y: number; width: number; height: number }
  ){
    return cornerPoints.every((point) => {
      const isWithinXRange =
        point.x >= scannableArea.x &&
        point.x <= scannableArea.x + scannableArea.width;
      const isWithinYRange =
        point.y >= scannableArea.y &&
        point.y <= scannableArea.y + scannableArea.height;
      return isWithinXRange && isWithinYRange;
    });
  };
```
This worked for me.

**Reply by matmartic** [reply]
@shel404 thanks a lot, this worked great

**Reply by plaknight** [reply]
Why is the origin of the bounds.origin in onBarCodeScanned located at the top right corner instead of the top left

**Reply by kumar11jr** [reply]
> ```js
>   const isWithinScannableArea = (
>     cornerPoints: Point[],
>     scannableArea: { x: number; y: number; width: number; height: number }
>   ){
>     return cornerPoints.every((point) => {
>       const isWithinXRange =
>         point.x >= scannableArea.x &&
>         point.x <= scannableArea.x + scannableArea.width;
>       const isWithinYRange =
>         point.y >= scannableArea.y &&
>         point.y <= scannableArea.y + scannableArea.height;
>       return isWithinXRange && isWithinYRange;
>     });
>   };
> ```
> 
> This worked for me.

This is working in portrait mode not in **landscape** mode. Is their any solution for landscape.

**Reply by Eonfuzz** [reply]
Looks like this breaks in latest camera version, unless you explicitly define aspect ratio the cornerPoints are incorrect and are offset by the amount the camera would be cropped by. I'm not certain if we can even access that data now
